### PR TITLE
drop node 14 support

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,31 +17,63 @@ on:
...
 
   test12:
     name: 'Run tests using Next@12'
     runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
 
     strategy:
       matrix:
-        node-version: [ 14, 16, 18 ]
```
Comment (SimenB) [inline .github/workflows/ci.yml:32]: Not removed here?

**Comment by Fdawgs** [comment]
Sorry been busy fighting ESM bullshit all day at work; is it worth bringing across the license checking, dependency review, and automerge jobs from the shared workflow too?

Just like `fluent-json-schema` has done https://github.com/fastify/fluent-json-schema/blob/master/.github/workflows/ci.yml

**Comment by Uzlopak** [comment]
@Fdawgs 

I hope you like this version ;)

@Eomm 
@SimenB 
@mcollina 
Significantly changed the PR.

**Review by SimenB — APPROVED** [review]
Nice! 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,32 +16,90 @@ on:
...
+      permissions:
+        contents: read
+      steps:
+          - name: Check out repo
+            uses: actions/checkout@v3
+            with:
+              persist-credentials: false
+
+          - name: Dependency review
+            uses: actions/dependency-review-action@v3
```
Comment (Eomm) [inline .github/workflows/ci.yml:32]: uuuu dangerous 😄

**Review by Eomm — APPROVED** [review]

**Pull request merged**